Model/GridSource.php

```diff
@@ -96,11 +94,8 @@ public function extractValue($record, string $key)
 
     private function getRawGridData(SearchCriteriaInterface $searchCriteria): RawGridSourceContainer
     {
-        if (!isset($this->rawGridData)) {
-            $preprocessedSearchCriteria = $this->preprocessSearchCriteria($searchCriteria);
-            $this->rawGridData          = $this->gridSourceType->fetchData($preprocessedSearchCriteria);
-        }
-        return $this->rawGridData;
+        $preprocessedSearchCriteria = $this->preprocessSearchCriteria($searchCriteria);
```

Comment:
I added the `rawGridData` memoization after profiling. I also don't like that it isn't bound to the search criteria values, but it still is important. It probably would be good to change the memoization to depend the search criteria values.

Reply:
I've changed that behavior to be able to iterate through all the pagination, otherwise it would be stuck within the current search criteria.

I thought that your idea behind that was to improve performance, but i think it would be better to cache this in the grid (because the grid knows its state). So probably it is better to do that in the getRows method. I've changed it to make the SearchCriteria optional. I think that should work best to keep both approaches.